### PR TITLE
Update react collapsible definitions to latest version

### PR DESCRIPTION
**Pull request by Stanley-Jovel**
# What does this PR do
It adds type definitions for the new props added in the most recent version of the library

### CODE REVIEW

File: definitions/npm/react-collapsible_v2.x.x/flow_v0.54.x-/react-collapsible_v2.x.x.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    triggerOpenedClassName?: string,
+    contentOuterClassName?: string,
+    contentInnerClassName?: string,
+    accordionPosition?: string | number,
+    handleTriggerClick?: (accordionPosition?: string | number) => void,
+    onOpen?: () => void,
+    onClose?: () => void,
+    onOpening?: () => void,
+    onClosing?: () => void,
+    trigger?: string | React$Node,
```
Comment (pascalduez) [inline definitions/npm/react-collapsible_v2.x.x/flow_v0.54.x-/react-collapsible_v2.x.x.js]: `React$Node` already allows strings.

Comment (pascalduez) [reply definitions/npm/react-collapsible_v2.x.x/flow_v0.54.x-/react-collapsible_v2.x.x.js]: From the `react-collapsible` documentation it's `string | React$Element<*>`

File: definitions/npm/react-collapsible_v2.x.x/flow_v0.54.x-/react-collapsible_v2.x.x.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    triggerWhenOpen?: string | React$Node,
```
Comment (pascalduez) [inline definitions/npm/react-collapsible_v2.x.x/flow_v0.54.x-/react-collapsible_v2.x.x.js]: From the `react-collapsible` documentation it's `string | React$Element<*>``

File: definitions/npm/react-collapsible_v2.x.x/flow_v0.54.x-/react-collapsible_v2.x.x.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    lazyRender?: boolean,
+    overflowWhenOpen?:
+      | "hidden"
+      | "visible"
+      | "auto"
+      | "scroll"
+      | "inherit"
+      | "initial"
+      | "unset",
+    triggerSibling?: React$Node | string | (() => void),
```
Comment (pascalduez) [inline definitions/npm/react-collapsible_v2.x.x/flow_v0.54.x-/react-collapsible_v2.x.x.js]: From the `react-collapsible` documentation it's `React$Element<*>`

**Comment by pascalduez** [comment]
I would double check with [their TS definitions](https://github.com/glennflanagan/react-collapsible/blob/develop/index.d.ts), quite some differences here.

**Comment by Stanley-Jovel** [comment]
@pascalduez Thank you for the feedback.
I made the suggested changes and after reading the docs now all props have the right typedef.

Also, after looking at the docs I also noticed that `trigger` prop is required (and actually the only required prop) so I made it required.

**Comment by pascalduez** [comment]
@Stanley-Jovel Thanks! Yeah tricky, there seem to be a desync between docs, code, and also their TS defs...

**Comment by GAntoine** [comment]
Merged with pascal's blessing and review, well done.

**Pull request merged**